components/brave_ads/core/browser/search_result_ad/search_result_ad_converting_util.cc

```diff
@@ -193,155 +191,131 @@ bool SetSearchAdProperty(const schema_org::mojom::PropertyPtr& ad_property,
-      return SearchResultAdMap();
+    const auto* it = base::ranges::find(kSearchResultAdAttributes,
+                                        base::StringPiece(ad_property->name));
+
+    // Unknown search result ad attribute specified.
+    if (it == kSearchResultAdAttributes.end()) {
+      continue;
     }
+    found_attributes.insert(*it);
 
-    for (const auto& ad_entity : property->values->get_entity_values()) {
-      if (!ad_entity || ad_entity->type != kSearchResultAdType) {
-        LOG(ERROR) << "Wrong search result ad type specified: "
-                   << ad_entity->type;
-        return SearchResultAdMap();
-      }
-
-      if (property->name == kTypePropertyName) {
-        continue;
-      }
-
-      SearchResultAdInfoPtr search_result_ad = SearchResultAdInfo::New();
-      search_result_ad->conversion = ConversionInfo::New();
-
-      base::flat_set<base::StringPiece> found_attributes;
-      for (const auto& ad_property : ad_entity->properties) {
-        // Wrong attribute name
-        const auto* it = base::ranges::find(
-            kSearchResultAdAttributes, base::StringPiece(ad_property->name));
-
-        if (it == kSearchResultAdAttributes.end()) {
-          LOG(ERROR) << "Wrong search result ad attribute specified: "
-                     << ad_property->name;
-          return SearchResultAdMap();
-        }
-        found_attributes.insert(*it);
-
-        if (!SetSearchAdProperty(ad_property, search_result_ad.get())) {
-          LOG(ERROR) << "Cannot read search result ad attribute value: "
-                     << ad_property->name;
-          return SearchResultAdMap();
-        }
-      }
-
-      // Not all of attributes were specified.
-      if (found_attributes.size() != kSearchResultAdAttributes.size()) {
-        std::vector<base::StringPiece> absent_attributes;
-        std::set_difference(kSearchResultAdAttributes.begin(),
-                            kSearchResultAdAttributes.end(),
-                            found_attributes.begin(), found_attributes.end(),
-                            std::back_inserter(absent_attributes));
-
-        LOG(ERROR) << "Some of search result ad attributes were not specified: "
-                   << base::JoinString(absent_attributes, ", ");
-
-        return SearchResultAdMap();
-      }
-
-      const std::string creative_instance_id =
-          search_result_ad->creative_instance_id;
-      search_result_ads.emplace(creative_instance_id,
-                                std::move(search_result_ad));
+    if (!SetSearchAdProperty(ad_property, search_result_ad.get())) {
+      VLOG(6) << "Cannot read search result ad attribute value: "
+              << ad_property->name;
+      return;
     }
+  }
+
+  // Not all of attributes were specified.
+  if (found_attributes.size() != kSearchResultAdAttributes.size()) {
+    std::vector<base::StringPiece> absent_attributes;
+    std::set_difference(kSearchResultAdAttributes.begin(),
```

Comment:
1. `base::ranges::set_difference`
2. wrap this thing into `if (VLOG_IS_ON(6)) {`

Reply:
Fixed